packages/shared/src/hooks/useContextMenu.ts

```diff
@@ -1,15 +1,18 @@
+import { useState } from 'react';
 import { useContextMenu as useContexifyContextMenu } from '@dailydotdev/react-contexify';
 
 export default function useContextMenu({ id }: { id: string }): {
   onMenuClick: (e: React.MouseEvent) => void;
 } {
+  const [menu, setMenu] = useState(true);
   const { show } = useContexifyContextMenu({ id });
 
   const onMenuClick = (e: React.MouseEvent) => {
+    !menu ? setMenu(true) : setMenu(false);
     const { right, bottom } = e.currentTarget.getBoundingClientRect();
-    show(e, {
-      position: { x: right, y: bottom + 4 },
-    });
+    menu && show(e, {
+        position: { x: right, y: bottom + 4 },
+      });
```

Comment:
I think the condition is the opposite. Also, let's also move my suggestion about `setMenu` to the last line of the this function.
```suggestion
    if (!menu) {
      show(e, {
        position: { x: right, y: bottom + 4 },
      });
    }
```